0022-circuit-block.md

```diff
@@ -0,0 +1,185 @@
+# Introducing `CircuitBlock`s in qiskit
+
+| **Status**        | **Proposed** |
+|:------------------|:---------------------------------------------|
+| **RFC #**         | 0022                                         |
+| **Authors**       | [Sam Ferracin](sam.ferracin@ibm.com), [Ian Hincks](ian.hincks@ibm.com), [Chris Wood](cjwood@ibm.com), [Jake Lishman](jake.lishman@ibm.com), [Joshua Skanes Norman](joshua.sn@ibm.com)    |
+| **Submitted**     | YYYY-MM-DD                                   |
+
+## Summary
+Objective: Introducing a new ``CircuitBlock`` class to enhance the user experience with Qiskit's built-in primitives, with the goal of providing more flexibility and full transparency with respect to twirling and mitigation.
+
+The concept of a "block" of gates—meaning a subset of gates isolated from the rest of the circuit—is fundamental to many quantum computing routines. However, Qiskit does not provide a dedicated object to represent a block. Traditionally, users have worked around this by defining blocks *indirectly* using barriers. While this approach has been adequate for tasks like scheduling and transpilation, barriers can be hard to parse when pre-processing circuits in preparation for twirling or mitigation. Suboptimal handling of barriers in these post-processing steps can introduce significant slowdowns in some mitigation experiments and produce outcomes that may seem unintuitive to typical users.
```

Comment:
I have some words that might somehow be edited into this paragraph:

A block of gates is a collection of circuit instructions that is promised to have a consistent execution when applied to a particular set of qubits. For example, the relative scheduling of gates within a block is consistent between multiple applications of the block. A circuit layer, or cycle, is a special case of a block that is full-width and depth-1.

In particular, a block is a natural concept when reasoning about quantum errors because the error profile often depends on factors such as which gates are applied simultaneously.

Reply:
I stole the second sentence. Feel free to add the first one too if you find a natural spot for it :)